morph_tool/nrnhines.py

```diff
@@ -273,3 +278,87 @@ def point_to_section_end(sections: Sequence[neuron.nrn.Section],
         if np.isclose(point, last_section_point, atol=atol, rtol=rtol).all():
             return index
     return None
+
+
+class NoDaemonProcess(multiprocessing.Process):
+    """Class that represents a non-daemon process"""
+
+    def _get_daemon(self):  # pylint: disable=R0201
+        """Get daemon flag"""
+        return False
+
+    def _set_daemon(self, value):
+        """Set daemon flag"""
+
+    daemon = property(_get_daemon, _set_daemon)
+
+
+class NestedPool(multiprocessing.pool.Pool):  # pylint: disable=abstract-method
+    """Class that represents a MultiProcessing nested pool"""
+
+    Process = NoDaemonProcess
+
+
+def isolate(func):
+    """Isolate a generic function for independent NEURON instances.
+
+    Note: it does not work as decorator.
+
+    Args:
+        func (function): function to isolate"""
+    def func_isolated(*args, **kwargs):
+        with NestedPool(1, maxtasksperchild=1) as pool:
+            return pool.apply(func, args, kwargs)
```

Comment:
```suggestion
            return pool.apply(func, *args, **kwargs)
```
I think this is the reason for the *morph_data

Reply:
this does not work, apply works like apply(func, args, kwargs) = func(*args, **kwargs), I tried several options, none work expect the current one.